cmd/kubeadm/app/util/config/common.go

```diff
@@ -93,19 +93,19 @@ func DetectUnsupportedVersion(b []byte) error {
 		}
 		knownKinds[gvk.Kind] = true
 	}
-	// InitConfiguration, MasterConfiguration and NodeConfiguration are mutually exclusive, error if more than one are specified
+
+	// InitConfiguration, MasterConfiguration and NodeConfiguration are mutually exclusive, warn if more than one are specified
 	mutuallyExclusive := []string{constants.InitConfigurationKind, constants.JoinConfigurationKind}
-	foundOne := false
+	mutuallyExclusiveCount := 0
 	for _, kind := range mutuallyExclusive {
 		if knownKinds[kind] {
-			if !foundOne {
-				foundOne = true
-				continue
-			}
-
-			return fmt.Errorf("invalid configuration: kinds %v are mutually exclusive", mutuallyExclusive)
+			mutuallyExclusiveCount++
 		}
 	}
+	if mutuallyExclusiveCount > 1 {
+		glog.Warningf("WARNING: configuration kinds %v are mutually exclusive", mutuallyExclusive)
```

Comment:
I wouldn't say mutually exclusive here, b/c if it were there would be an error.  

Maybe 
"Detected resource kinds that may not apply: %v"